### PR TITLE
[ZEPPELIN-996] Improve first page and dropdown menu in documentation site

### PR DESCRIPTION
**Pull request by AhyoungRyu**
### What is this PR for?
Current Zeppelin documentation site is little bit hard to find a way for Zeppelin beginners. It will not easy  to improve this at a time, but I did the below as a start of this work.

1. Restructured dropdown menu and added each category names
2. Added a overview list(with short description) to first page of website (index.md) so that people can look through the overall contents in Zeppelin website at a glance. (as [Apache Spark](http://spark.apache.org/docs/latest/#where-to-go-from-here) and [Apache Mesos](http://mesos.apache.org/documentation/latest/) does)

Please see the attached screenshot images :)

### What type of PR is it?
Improvement & Documentation

### Todos
* [x] - Change outdated screenshot images 
* [x] - Combine `text.md`, `table.md` and `html.md` to `basicdisplaysystem.md`
* [x] - Fix dead link in `virtual_machine.md`
* [x] - Improve dropdown menu and reorder
* [x] - Improve first page(`index.md`)
* [x] - Combine with #995 after it is merged into master

### What is the Jira issue?
[ZEPPELIN-996](https://issues.apache.org/jira/browse/ZEPPELIN-996)

### How should this be tested?
1. Apply this patch and [build the docs website with jekyll](https://github.com/apache/zeppelin/tree/master/docs#build-documentation)
2. Check the first page(index.html) and dropdown menu

### Screenshots (if appropriate)
 - Dropdown menu
dropdown

 - First page
screen shot 2016-06-14 at 1 28 58 pm
screen shot 2016-06-14 at 1 29 11 pm
screen shot 2016-06-14 at 1 29 39 pm
screen shot 2016-06-14 at 1 29 57 pm
screen shot 2016-06-14 at 1 30 12 pm
screen shot 2016-06-14 at 1 30 22 pm

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by corneadoug** [comment]
Great work,
Maybe in the case of the interpreter menu, we should bring the usage section before the available interpeters (since there is only one item)

**Comment by AhyoungRyu** [comment]
@corneadoug Ah there is two menu actually, they are not shown properly in the screenshot. 
 - Dynamic Interpreter Loading 
 - Dependency Management 

Anyway, I can change the placement as you suggested :)

**Comment by corneadoug** [comment]
Oh indeed,
Since they are more generalistic and also a but hidden down there, it could be a better read flow to have them right below the overview

**Comment by AhyoungRyu** [comment]
@corneadoug Yeah you're right and it makes sense indeed. I addressed it :)

File: docs/displaysystem/basicdisplaysystem.md [hunk 1]
```diff
@@ -18,19 +18,36 @@ See the License for the specific language governing permissions and
 limitations under the License.
 -->
 {% include JB/setup %}
+## Html
+
+With `%html` directive, Zeppelin treats your output as html
```
Comment (bzz) [inline docs/displaysystem/basicdisplaysystem.md]: A nitpick - it's probably 'as HTML' capital case, as it's referred below

File: docs/displaysystem/basicdisplaysystem.md [hunk 2]
```diff
@@ -18,19 +18,36 @@ See the License for the specific language governing permissions and
...
 Note that display system is backend independent.
 
-If table contents start with %html, it is interpreted as an HTML.
+If table contents start with `%html`, it is interpreted as an HTML.
 
 <img src="/assets/themes/zeppelin/img/screenshots/display_table_html.png" />
+
+## Text
+
+Zeppelin prints output of language backend in text, by default.
```
Comment (bzz) [inline docs/displaysystem/basicdisplaysystem.md]: By default, Apache Zeppelin prints interpreter responce as a plain text using `text` display system.

File: docs/displaysystem/basicdisplaysystem.md [hunk 3]
```diff
@@ -18,19 +18,36 @@ See the License for the specific language governing permissions and
...
+
+<img src="/assets/themes/zeppelin/img/screenshots/display_text.png" />
+
+You can explicitly say you're using text display system.
```
Comment (bzz) [inline docs/displaysystem/basicdisplaysystem.md]: It is possible to explicitly specify display system as `text`

Comment (bzz) [inline docs/displaysystem/basicdisplaysystem.md]: It is possible to explicitly specify display system as `text` as well

Comment (bzz) [inline docs/displaysystem/basicdisplaysystem.md]: It is also possible to explicitly specify display system as `text`

File: docs/displaysystem/basicdisplaysystem.md [hunk 4]
```diff
@@ -18,19 +18,36 @@ See the License for the specific language governing permissions and
...
+
+<img src="/assets/themes/zeppelin/img/screenshots/display_text1.png" />
+
+Note that display system is backend independent.
```
Comment (bzz) [inline docs/displaysystem/basicdisplaysystem.md]: I think exactly the same sentence is already present above

File: docs/index.md [hunk 5]
```diff
@@ -74,49 +71,104 @@ Some basic charts are already included in Zeppelin. Visualizations are not limit
...
-Learn more about Zeppelin's Display system. ( [text](./displaysystem/display.html), [html](./displaysystem/display.html#html), [table](./displaysystem/table.html), [angular](./displaysystem/angular.html) )
+
+Learn more about [display systems](#display-system) in Apache Zeppelin.
 
 <br />
-### Dynamic forms
+## Dynamic forms
 
-Zeppelin can dynamically create some input forms into your notebook.
+You can dynamically create some input forms in your notebook.
```
Comment (bzz) [inline docs/index.md]: Generally, previous writing style of "Zeppelin can ..." is prefered to the one that is dominant in this PR, like "You can ...", "Your notebooks ..." - first one feels more like a documentation about the project.

What do you think, is it possible\make sense to adjust it here?

File: docs/displaysystem/basicdisplaysystem.md (hunk 4 above)
Comment (AhyoungRyu) [inline docs/displaysystem/basicdisplaysystem.md]: Thanks for pointing out them. I should've read overall contents again :)

File: docs/index.md (hunk 5 above)
Comment (AhyoungRyu) [inline docs/index.md]: Sure why not. It makes sense indeed. Thanks!

**Comment by AhyoungRyu** [comment]
@bzz Thanks for your prudent review! I just addressed them in [6a2f40b](https://github.com/apache/zeppelin/pull/1004/commits/6a2f40b73e65986bdd61b611c76243f7a8bd7916).

**Comment by minahlee** [comment]
screen shot 2016-06-14 at 10 34 51 am

This is what I see from my screen, and cannot scroll down lower. Can you adjust size of the dropdown menu?

And also, can we rollback the change of display system menu order to text -> html -> table? if there are a lot of menu then alphabetical order makes sense, but in this case I think better to display in order of basic to advanced

Other than above comments, this PR looks awesome!

**Comment by AhyoungRyu** [comment]
@minahlee Thanks for your prudent feedback. I just addressed them to my last some commits as you said. And I also applied scrollbar to "More" tab. 

BTW, I found an UI issue on "More" tab. When the screen width is small, some long text in the list are not visible. It was also fixed at [af70939](https://github.com/apache/zeppelin/pull/1004/commits/af70939b4231101e7f3760b6367a05704d3912c4)

 - Before 
screen shot 2016-06-14 at 1 36 10 pm

 - After
screen shot 2016-06-14 at 1 36 58 pm

**Comment by minahlee** [comment]
merging if there is no more discussion

**Pull request closed without merging**